toml/src/test/kotlin/org/rust/toml/CargoCrateDocLineMarkerProviderTest.kt

```diff
@@ -70,4 +71,28 @@ class CargoCrateDocLineMarkerProviderTest : CargoTomlLineMarkerProviderTestBase(
         [dependencies]
         base64 = "=0.8.0"  # - Open documentation for `base64@=0.8.0`
     """)
+
+    fun `test docs link with semver major`() {
+        val link = CargoCrateDocLineMarkerProvider.prepareDocsLink("abc", "1");
+        assertEquals("https://docs.rs/abc/%5E1", link);
+        assertNotNull(URL(link));
```

Comment:
I've created some infrastructure to check generated URLs in #7231. It allows creating integration tests instead of unit ones and not expose implementation details